### DISCUSSION TITLE
60 fps: What Are The Specifics?

### DISCUSSION THREAD

**Discussion by Bonasomnium** [ScratchAddons/ScratchAddons#5922] [Help] [answered]
So, I have a few questions regarding the 60 fps option

1: should I program my game to be made for 60fps mode if I want to? In that case, should I tell people to get Scratch addons to enable it or would it work just fine? (I assume not)

2: what about Turbowarp? If people didn't want to get scratch add-ons, could they use Turbowarp instead?

that's pretty much all the questions I have

**Comment by WorldLanguages** [comment, marked as answer]
> should I program my game to be made for 60fps mode if I want to

Theoretically you could detect 60FPS mode, but I wouldn't encourage you to only support 60FPS. You could support both 30FPS and 60FPS.

> should I tell people to get Scratch addons to enable it or would it work just fine?

Depends, you shouldn't do that within the Scratch website because it's not allowed to reference/advertise browser extensions.

> what about Turbowarp? If people didn't want to get scratch add-ons, could they use Turbowarp instead?

Sure. TurboWarp has 60FPS with interpolation which works better than Scratch Addons' 60FPS, and you don't have to have it in mind when developing your project because it will just work. Our 60FPS addon runs everything twice as fast, while TurboWarp's interpolation runs the project at normal speed but shows it with twice the framerate. Also, you can reference TurboWarp on Scratch without any problems.

**Reply by Bonasomnium** [reply]
Thank you so much! I was prepared to wait a week for an answer!
So, could you elaborate on the difference between Turbowarp and 60fps mode?

**Reply by DNin01** [reply]
Basically, TurboWarp has both a 60FPS mode that is the same as Scratch Addons but also comes with an alternative to it - Interpolation. One simply speeds up scripts and one doesn’t.

**Reply by Bonasomnium** [reply]
Thanks! I mostly was torn between 60fps mode and normal because it looked smooth, but some of my scripts relied on the time it takes scripts to load to function as intended. _May your code work on the first try!_

**Comment by DNin01** [comment]
I’d also like to note that you should probably aim to mainly support 30FPS if you want to share your project on Scratch, since a lot of users aren’t going to take links to TurboWarp, and if you ever plan on only supporting 60FPS, then it may be a good idea to share it elsewhere, since sharing to Scratch means that your project should work on Scratch.

**Reply by Bonasomnium** [reply]
good point! I think I'll stick to it, but advertise 60fps mode as higher quality!
